include/class.dynamic_forms.php

```diff
@@ -1357,13 +1357,17 @@ function getField() {
     }
 
     function getValue() {
-
+        $field = $this->getField();
         if (!isset($this->_value)) {
             //XXX: We're settting the value here to avoid infinite loop
             $this->_value = false;
-            if (isset($this->value))
-                $this->_value = $this->getField()->to_php(
+            if (isset($this->value)) {
+                if ($field->get('type') == 'choices')
```

Comment:
I don't like this - why got do `$field->getValue()` and abstract dirty work to the class itself?

Reply:
My comment refers to it as `$field` not `$this` - anyhow - I'll reimplement the fix then let you see what I meant.